### PR TITLE
feat: can define ontology labels

### PR DESCRIPTION
**Pull request by mswertz**
Motivation: 
- sometimes same names must be used in multiple sub hierarchies; this would require prefix. To keep UI clean you can now use a shorter label.

For example:
- prescriptions:xyz 
- dispensings:xyz
in both places you would like to show 'xyz' on in the form. (in the filter statements you would like to see the prescriptions:xyz)

N.B. only works on newly created ontology tables; do we want a migration?

### CODE REVIEW

**Comment by mswertz** [comment]
[As ontology admin I want labels so I can have same term multiple times (without having stupid prefixes).](https://trello.com/c/hQ01fMQY/577-as-ontology-admin-i-want-labels-so-i-can-have-same-term-multiple-times-without-having-stupid-prefixes)

**Review by mswertz — COMMENTED** [review]
We noticed a bug in the user interface: filters didn't work properly

**Review by BrendaHijmans — COMMENTED** [review]
funtionality works the way we want it, cannot comment on code though. i want to ask joeri if he sees any issues with this.

File: backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlColumnExecutor.java [hunk 1]
```diff
@@ -287,26 +287,33 @@ public static TableMetadata getOntologyTableDefinition(String name, String descr
...
                 .setPkey()
                 .setRequired(true)
-                .setDescription("Short but user-friendly name for this term")
+                .setDescription("Unique name of the term within this table")
                 .setSemantics("http://purl.obolibrary.org/obo/NCIT_C42614"),
+            column("label")
+                // .setKey(2) when we upgrade to psql 15 so we can allow parent == null in
+                // constraint so we can ensure unique labels on each level
+                .setDescription("User-friendly label for this term. Should be unique in parent")
+                .setSemantics("http://purl.obolibrary.org/obo/NCIT_C42614"),
```
Comment (joerivandervelde) [inline backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlColumnExecutor.java:296]: suggested term to use here: http://purl.obolibrary.org/obo/NCIT_C45561

File: apps/molgenis-components/src/components/forms/InputOntology.vue [hunk 2]
```diff
@@ -558,8 +560,8 @@ export default {
           description="please choose your options in tree below"
           v-model="value"
           :isMultiSelect="false"
-          tableName="Tag"
```
Comment (joerivandervelde) [inline apps/molgenis-components/src/components/forms/InputOntology.vue:561]: has the demo been changes from Pet Store to a Catalogue?

**Review by joerivandervelde — CHANGES_REQUESTED** [review]

**Comment by joerivandervelde** [comment]
one last remark: EMX2 docs should say that label is optional and name will be used if no label has been filled in, and that label can be used to denote that a term is used multiple times within an 'application ontology', where name then functions as the path of this term within the application ontology

File: apps/molgenis-components/src/components/forms/InputOntology.vue (hunk 2 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/InputOntology.vue:561]: thx for spotting need to undo

**Comment by mswertz** [comment]
> one last remark: EMX2 docs should say that label is optional and name will be used if no label has been filled in, and that label can be used to denote that a term is used multiple times within an 'application ontology', where name then functions as the path of this term within the application ontology

done

**Review by joerivandervelde — APPROVED** [review]

**Pull request merged**